### PR TITLE
Delta.

### PR DESCRIPTION
**Pull request by mjmacleod**
Backport delta from 0.10 branch in anticipation of release.
Adjust drift.
Add new checkpoints.

### CODE REVIEW

File: src/diff_delta.h [hunk 1]
```diff
@@ -0,0 +1,321 @@
...
+    // How many blocks to use for the fifth window, fifth window is across all algorithms.
+    #ifdef MULTI_ALGO
+    const int nDayFrame            = 576;
+    #endif
+
+    // Weighting to use for each of the four algo specific time windows.
+    const int64_t nLBWeight        =  64;
+    const int64_t nShortWeight     =  8;
+    int64_t nMiddleWeight          =  2;
+    int64_t nLongWeight            =  1;
```
Comment (GeertJohan) [inline src/diff_delta.h]: Just curious, why are these not `const`?

File: src/main.cpp [hunk 2]
```diff
@@ -1387,17 +1387,20 @@ unsigned int static GetNextWorkRequired_DIGI(const CBlockIndex* pindexLast, cons
     return bnNew.GetCompact();
 }
 
-
+#include "diff_delta.h"
 unsigned int static GetNextWorkRequired(const CBlockIndex* pindexLast, const CBlockHeader *pblock)
 {
+	static int nDeltaSwitchover = fTestNet ? 100 : 213500;
```
Comment (GeertJohan) [inline src/main.cpp]: I like this, can we expand this to all diff algo's? `nDGW3Switchover`, `nDigiSwitchover`, `nKGWSwitchover`...

File: src/diff_delta.h (hunk 1 above)
Comment (mjmacleod) [inline src/diff_delta.h]: There is a special case (if more than four blocks come in faster than X
and the latest block is 'under time') where we ignore the middle and
long time frames entirely - these variables need to be set to 0 for that
case.

File: src/main.cpp (hunk 2 above)
Comment (mjmacleod) [inline src/main.cpp]: Yeah we could do it for new codebase, probably no point in doing it for
old codebase at this point I guess.

File: src/diff_delta.h (hunk 1 above)
Comment (GeertJohan) [inline src/diff_delta.h]: Ah, I see, yes. Looks pretty neat!

File: src/main.cpp (hunk 2 above)
Comment (GeertJohan) [inline src/main.cpp]: :+1:

**Pull request merged**